python_modules/dagster/dagster/_config/pythonic_config/__init__.py

```diff
@@ -1784,11 +1749,23 @@ def separate_resource_params(data: Dict[str, Any]) -> SeparatedResourceParams:
 def _call_resource_fn_with_default(
     stack: contextlib.ExitStack, obj: ResourceDefinition, context: InitResourceContext
 ) -> Any:
+    from dagster._config.validate import process_config
+
     if isinstance(obj.config_schema, ConfiguredDefinitionConfigSchema):
         value = cast(Dict[str, Any], obj.config_schema.resolve_config({}).value)
         context = context.replace_config(value["config"])
     elif obj.config_schema.default_provided:
-        context = context.replace_config(obj.config_schema.default_value)
+        unprocessed_config = obj.config_schema.default_value
```

Comment:
Oh man good catch.

This is worth an extended comment explaining how by the time we get here:

*  `context.resource_config` has _already_ been post processed.
*  The nested resource's config has _also_ been post processed. However it is not available at this point in the stack. The nested resources's post processed config is in the broader run_config dictionary. (e.g. it would live somewhere in process as `run_config['resources']['nested_resource']` if the nested resource had key `nested_resource`
* The only way we have access to the information here is through the unprocessed default value, so in the current state of things we have to get the default value, process it, and jam it into the resource config.